mteb/cache.py

```diff
@@ -214,3 +223,249 @@ def clear_cache(self) -> None:
 
     def __repr__(self) -> str:
         return f"ResultCache(cache_path={self.cache_path})"
+
+    def get_cache_paths(
+        self,
+        models: Sequence[str] | Sequence[ModelMeta] | None = None,
+        tasks: Sequence[str] | Sequence[AbsTask] | None = None,
+        require_model_meta: bool = True,
+        include_remote: bool = True,
+    ) -> list[Path]:
+        """Get all paths in the cache directory.
+
+        Args:
+            models: A list of model names or ModelMeta objects to filter the paths.
+            tasks: A list of task names to filter the paths.
+            require_model_meta: If True, only return paths that have a model_meta.json file.
+            include_remote: If True, include remote results in the returned paths.
+
+        Returns:
+            A list of paths in the cache directory.
+
+        Example:
+            >>> from mteb.cache import ResultCache
+            >>> cache = ResultCache()
+            >>>
+            >>> # Get all cache paths
+            >>> paths = cache.get_cache_paths()
+            >>>
+            >>> # Get all cache paths for a specific task
+            >>> paths = cache.get_cache_paths(tasks=["STS12"])
+            >>>
+            >>> # Get all cache paths for a specific model
+            >>> paths = cache.get_cache_paths(models=["sentence-transformers/all-MiniLM-L6-v2"])
+            >>>
+            >>> # Get all cache paths for a specific model and revision
+            >>> model_meta = mteb.get_model_meta("sentence-transformers/all-MiniLM-L6-v2")
+            >>> paths = cache.get_cache_paths(models=[model_meta])
+        """
+        cache_paths = [
+            p
+            for p in (self.cache_path / "results").glob("**/*.json")
+            if p.name != "model_meta.json"
+        ]
+        if include_remote:
+            cache_paths += [
+                p
+                for p in (self.cache_path / "remote").glob("**/*.json")
+                if p.name != "model_meta.json"
+            ]
+
+        cache_paths = self._filter_paths_by_model_and_revision(
+            cache_paths,
+            models=models,
+        )
+        cache_paths = self._filter_paths_by_task(cache_paths, tasks=tasks)
+
+        if require_model_meta:
+            cache_paths = [
+                p for p in cache_paths if (p.parent / "model_meta.json").exists()
+            ]
+        return cache_paths
+
+    def get_models(
+        self,
+        tasks: Sequence[str] | None = None,
+        require_model_meta: bool = True,
+        include_remote: bool = True,
+    ) -> list[tuple[ModelName, Revision]]:
+        """Get all models in the cache directory."""
+        cache_paths = self.get_cache_paths(
+            tasks=tasks,
+            require_model_meta=require_model_meta,
+            include_remote=include_remote,
+        )
+        models = [(p.parent.parent.name, p.parent.name) for p in cache_paths]
+        return models
+
+    def get_tasks(
```

Comment:
Could we try to avoid confusion with `mteb.get_tasks`, which returns a `MTEBTasks` object?
```suggestion
    def get_task_names(
```